packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,50 @@
+import qs from 'query-string';
+
+import { withReservedScopes } from '../utils/scopes';
+
+const prompt = 'consent';
+const codeChallengeMethod = 'S256';
+
+enum ResponseType {
+  code = 'code',
+}
```

Comment:
So why is response type defined as an enum, while the other two above are constants? Use constants for all?

Reply:
- `prompt` is not suitable to define as enum for now
- `codeChallengeMethod` is fixed (defined by RFC)
- since more response types will be used in the near future, i prefer to predefine `ResponseType` (but define the only used one for now) (will be exported and moved to suitable place)

Reply:
How about adding it when we really need it? Just took a look at swift sdk and found Gao was using string variable.

Reply:
Updated, PTAL.